pkg/reconciler/taskrun/taskrun.go

```diff
@@ -106,11 +106,15 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 	// If the TaskRun is just starting, this will also set the starttime,
 	// from which the timeout will immediately begin counting down.
-	tr.Status.InitializeConditions()
-	// In case node time was not synchronized, when controller has been scheduled to other nodes.
-	if tr.Status.StartTime.Sub(tr.CreationTimestamp.Time) < 0 {
-		c.Logger.Warnf("TaskRun %s createTimestamp %s is after the taskRun started %s", tr.GetRunKey(), tr.CreationTimestamp, tr.Status.StartTime)
-		tr.Status.StartTime = &tr.CreationTimestamp
+	if !tr.HasStarted() {
+		tr.Status.InitializeConditions()
+		// In case node time was not synchronized, when controller has been scheduled to other nodes.
```

Comment:
is there anywhere else we make this kind of comparision? this seems like the sort of thing that might be well suited to a separate package with unit tests (e.g. a package that allows us to compare times safely in cases of rescheduling)

Reply:
We do the same in the pipeline controller.
I agree with your comment, but I would prefer to do that change in a PR on top.